inc/log.class.php

```diff
@@ -704,6 +823,367 @@ static function getHistoryData(CommonDBTM $item, $start = 0, $limit = 0, array $
       return $changes;
    }
 
+   /**
+    * Retrieve distinct values for user_name field in item log.
+    * Return is made to be used as select tag options.
+    *
+    * @param CommonDBTM $item  Object instance
+    *
+    * @return array
+    **/
+   static function getDistinctUserNamesValuesInItemLog(CommonDBTM $item) {
+      global $DB;
+
+      $itemtype = $item->getType();
+      $items_id = $item->getField('id');
+
+      $iterator = $DB->request([
+         'SELECT DISTINCT' => 'user_name',
+         'FROM'            => self::getTable(),
+         'WHERE'  => [
+               'items_id'  => $items_id,
+               'itemtype'  => $itemtype
+            ],
+         'ORDER'  => 'id DESC'
+      ]);
+
+      $values = [];
+      while ($data = $iterator->next()) {
+         if (empty($data['user_name'])) {
+            continue;
+         }
+         $values[$data['user_name']] = $data['user_name'];
+      }
+
+      asort($values, SORT_NATURAL | SORT_FLAG_CASE);
+
+      return $values;
+   }
+
+   /**
+    * Retrieve distinct values for affected field in item log.
+    * Return is made to be used as select tag options.
+    *
+    * @param CommonDBTM $item  Object instance
+    *
+    * @return array
+    **/
+   static function getDistinctAffectedFieldValuesInItemLog(CommonDBTM $item) {
+      global $DB;
+
+      $itemtype = $item->getType();
+      $items_id = $item->getField('id');
+
+      $iterator = $DB->request([
+         'SELECT DISTINCT' => ['linked_action', 'itemtype_link', 'id_search_option'],
```

Comment:
not sure distinct will work as expected on several fields

Reply:
You are right. I updated in commit f34462212c26558b8c52df65266ef7db60458c7e it to use a `GROUP BY`.